### PR TITLE
Add pjproject 2.12.1

### PR DESCRIPTION
**Pull request by silentindark**
More information:
https://github.com/pjsip/pjproject/releases/tag/2.12.1

### CODE REVIEW

**Comment by gtjoseph** [comment]
We have to test before accepting 2.12.1.
https://issues.asterisk.org/jira/browse/ASTERISK-30050

**Comment by gtjoseph** [comment]
Actually, I'm going to re-open this PR but can you squash down the commits into 1?

**Comment by silentindark** [comment]
@gtjoseph i really have no idea how to do this

**Comment by seanbright** [comment]
@gtjoseph you can squash it when merging the pull request. The 'merge pull request' button has a drop down with a 'squash and merge' option.

**Comment by gtjoseph** [comment]
> @gtjoseph you can squash it when merging the pull request. The 'merge pull request' button has a drop down with a 'squash and merge' option.

Yeah but I'd rather see the final result before I merge and @silentindark 's fork will be out of date.

@silentindark Here's what you can do...
In your local repo...
$ git rebase -i HEAD~10
You should see something like this in your editor...
```
pick 295e8e3 Add pjproject 2.12.1
pick db8a9b5 Add pjproject 2.12
pick a33b7e1 Add pjproject 2.12.1 # empty
pick c58f447 Add pjproject 2.12.1
pick 139dff6 Add pjproject 2.12.1
pick 2eaa736 Delete SHA256SUM.TXT
pick d63362d Add pjproject 2.12.1
pick a07598b Add pjproject 2.12.1
pick a7f3cb0 Create test
pick fda9980 Delete test

# Rebase 5165c78..fda9980 onto 5165c78 (10 commands)
#
# Commands:
# p, pick <commit> = use commit
# r, reword <commit> = use commit, but edit the commit message
# e, edit <commit> = use commit, but stop for amending
# s, squash <commit> = use commit, but meld into previous commit
# f, fixup [-C | -c] <commit> = like "squash" but keep only the previous
#                    commit's log message, unless -C is used, in which case
#                    keep only this commit's message; -c is same as -C but
#                    opens the editor
...
```
Delete the two lines that create and delete the test and change "pick" to "fixup" (or just "f") for all but the first line...
```
pick 295e8e3 Add pjproject 2.12.1
f db8a9b5 Add pjproject 2.12
f a33b7e1 Add pjproject 2.12.1 # empty
f c58f447 Add pjproject 2.12.1
f 139dff6 Add pjproject 2.12.1
f 2eaa736 Delete SHA256SUM.TXT
f d63362d Add pjproject 2.12.1
f a07598b Add pjproject 2.12.1

# Rebase 5165c78..fda9980 onto 5165c78 (10 commands)
```
Save and close your editor.
You should now have only 1 commit that adds the checksum and archives.
Do a `git push --force` and it will update your fork and update the pull request at the same time.

**Comment by silentindark** [comment]
@gtjoseph Thank you so much for help!

**Comment by seanbright** [comment]
> > @gtjoseph you can squash it when merging the pull request. The 'merge pull request' button has a drop down with a 'squash and merge' option.
> 
> Yeah but I'd rather see the final result before I merge and @silentindark 's fork will be out of date.

That's what the [_Files changed_ tab](https://github.com/asterisk/third-party/pull/11/files) is for. I don't follow the "out of date" comment.

File: pjproject/2.12.1/MD5SUM.TXT [hunk 1]
```diff
@@ -0,0 +1,3 @@
+7f80ba8e1540853f959be6be7912a150  pjproject-2.12.1.tar.bz2
+339fcf1373276184fbdb821f75a1efbf  pjproject-2.12.1.tar.gz
```
Comment (gtjoseph) [inline pjproject/2.12.1/MD5SUM.TXT:2]: The checksums for the tar.gz and zip files don't match what I get...
```
a0250481bb082d553ae671bee7519d1b  pjproject-2.12.1.tar.gz
94823f8556403b72e47486089c1cad96  pjproject-2.12.1.zip
```
The tar.bz2 is OK as are the sha256 checksums.

**Review by gtjoseph — CHANGES_REQUESTED** [review]

File: pjproject/2.12.1/MD5SUM.TXT [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+e086683e3441be6e609fc4ce8cf2757f  pjproject-2.12.1.zip
```
Comment (gtjoseph) [inline pjproject/2.12.1/MD5SUM.TXT]: The checksums for the tar.gz and zip files don't match what I get when I run md5sum...
```
a0250481bb082d553ae671bee7519d1b  pjproject-2.12.1.tar.gz
94823f8556403b72e47486089c1cad96  pjproject-2.12.1.zip
```
The bz2 and sha256 checksums are OK though.

**Review by gtjoseph — CHANGES_REQUESTED** [review]

**Comment by gtjoseph** [comment]
@seanbright This is what I meant.  It would have been hard to make changes if @silentindark hadn't squashed the commits on his end because the commits added, deleted then added files again.   He doesn't need to squash the commit to fix the checksums though because at least now there's a good base commit for the pull request.  I can squash them as I merge them.

**Comment by silentindark** [comment]
@gtjoseph You are right, I fixed checksums for the tar.gz and zip files
Sorry, I don't know how it happened.
I did changes and now this commit as separate commit, it is suitable? Or I should again squash the commits into 1?

**Comment by gtjoseph** [comment]
> 



> I did changes and now this commit as separate commit, it is suitable? Or I should again squash the commits into 1?

You're good. Squashing the original commits created a good base so this one I can squash as I merge.

**Review by gtjoseph — APPROVED** [review]
Much better, thanks!

**Pull request merged**